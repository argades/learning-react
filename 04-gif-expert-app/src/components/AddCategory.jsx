import PropTypes from 'prop-types';
import { useState } from 'react';

export const AddCategory = ({ onNewCategory }) => {

  const [inputValue, setInputValue] = useState('');

  const onChange = ({target}) => {
    setInputValue(target.value);
  }

  const onSubmit = (event) => {
    event.preventDefault();
    const newInpVal = inputValue.trim();
    if ( newInpVal.length <=1 ) return;

    onNewCategory(newInpVal);

    setInputValue('');
  }

  return (
    <>
      <form onSubmit={ onSubmit } aria-label='form'>
        <input id="inputCategoria"
          type="text"
          placeholder="CategorÃ­a a buscar..." 
          value={ inputValue }
          onChange={ onChange }/>
      </form>
    </>
  )
}

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired,
}