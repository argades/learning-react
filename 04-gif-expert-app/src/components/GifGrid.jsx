import { useEffect } from "react";
import { getGif } from "../helper/getGifs";

/**
 * 
 * @param {*} param0 
 * @returns 
 */
export const GifGrid = ( { category } ) => {

  useEffect( () => {
    getGif(category);

  }, []);

  return (
    <>
      <h1>{ category }</h1>
    </>
  )
}