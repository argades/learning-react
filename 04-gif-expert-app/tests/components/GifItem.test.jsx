import { render, screen } from '@testing-library/react';
import { GifItem } from '../../src/components/GifItem';

describe('Testing <GifItem/>', () => { 
  const title = 'Title gif';
  const url = 'http://thunder-cats.com/alan';
  
  test('should test title and url required', () => {

    const { container } = render( <GifItem /> );
    expect( container ).toMatchSnapshot();
  });

  test('should test snapshot', () => {
    
    const { container } = render( <GifItem title={ title } url={ url }/> );
    expect( container ).toMatchSnapshot();
  });

  test('should show image with url and ALT', () => {
    render( <GifItem title={ title } url={ url }/> );
    //Para ver el objeto renderizado y poder extraer lo necesario para las validaciones
    //screen.debug(); 
    // expect( screen.getByRole('img').src ).toBe( url );
    // expect( screen.getByRole('img').alt ).toBe( title );

    // Forma optima para validar m√°s propiedades
    const { src, alt } = screen.getByRole('img');
    expect( src ).toBe( url );
    expect( alt ).toBe( title );

  });
  
  
 })