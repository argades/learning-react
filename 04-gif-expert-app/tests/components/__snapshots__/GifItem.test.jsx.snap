// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <GifItem/> should test snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <p>
      Title gif
    </p>
    <img
      alt="Title gif"
      src="http://thunder-cats.com/alan"
    />
  </div>
</div>
`;

exports[`Testing <GifItem/> should test title and url required 1`] = `
<div>
  <div
    class="card"
  >
    <p />
    <img />
  </div>
</div>
`;
